docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,804 @@
+
+The `Auth0` Login Flow diagramm should look like this now:
+
+image::auth0-login-flow.png[Auth0 Login Flow]
+
+Now you need to add a role such as `admin` to the users registered in the `Auth0` application.
+
+Create an `admin` role:
+
+image::auth0-create-role.png[Auth0 Create Role]
+
+and add it to the registered user:
+
+image::auth0-add-role-to-user.png[Auth0 Add Role to User]
+
+Next, update the Quarkus endpoint to require that only users with the `admin` role can access the endpoint:
+
+[source,java]
+----
+package org.acme;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import io.quarkus.oidc.IdToken;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @Inject
+    @IdToken
+    JsonWebToken idToken;
+	
+    @GET
+    @RolesAllowed("admin")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello, " + idToken.getClaim("name");
+    }
+    
+    @GET
+    @Path("post-logout")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String postLogout() {
+        return "You were logged out";
+    }
+}
+----
+
+Open `http://localhost:8080/hello`, authenticate to `Auth0` and get `403`. The reason you get `403` is because Quarkus OIDC does not know which claim in the `Auth0` tokens represents the roles information, by default a `groups` claim is checked, while `Auth0` tokens are now expected to have an "https://quarkus-security.com/roles" claim.
+
+We can fix it by telling Quarkus OIDC which claim must be checked to enforce RBAC:
+
+[source,configuration]
+----
+quarkus.oidc.auth-server-url=https://dev-3ve0cgn7.us.auth0.com
+quarkus.oidc.application-type=hybrid
+quarkus.oidc.authentication.scopes=profile
+quarkus.oidc.client-id=sKQu1dXjHB6r0sra0Y1YCqBZKWXqCkly
+quarkus.oidc.credentials.secret=${client-secret}
+
+quarkus.oidc.roles.role-claim-path="https://quarkus-security.com/roles" <1>
+
+# Logout
+quarkus.oidc.end-session-path=v2/logout
+quarkus.oidc.logout.post-logout-uri-param=returnTo
+quarkus.oidc.logout.extra-params.client_id=${quarkus.oidc.client-id}
+quarkus.oidc.logout.path=/logout
+quarkus.oidc.logout.post-logout-path=/hello/post-logout
+quarkus.http.auth.permission.authenticated.paths=/logout
+quarkus.http.auth.permission.authenticated.policy=authenticated
+----
+<1> Point to the custom roles claim. The path to the roles claim is in double quotes because the claim is namespace qualified.
+
+Now, clear the browser cookie cache just in case, open `http://localhost:8080/hello` again, authenticate to `Auth0` and get an expected user name.
+
+== Access Quarkus with Auth0 access tokens
```

Comment:
Note somewhere that if the access token is also in the JWT format then it can be verified directly